tests/test_package_registry_cargo.py

```diff
@@ -0,0 +1,42 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+
+# This software is licensed under the Apache License version 2.0.
+
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import json
+from unittest import mock
+
+from fetchcode.package import info
+
+
+def file_data(file_name):
+    with open(file_name) as file:
+        data = file.read()
+        return data
+
+
+@mock.patch("fetchcode.package.get_response")
+def test_cargo_packages(mock_get):
+    mock_get.return_value = json.loads(file_data("tests/data/cargo_mock_data.json"))
+    packages = list(info("pkg:cargo/rand"))
+    data = {}
+    for idx in range(0, len(packages)):
```

Comment:
This whole transformation of data could be much simplified to this:
`data = [p.to_dict() for p in packages]`
and this can then simply checked directly against the results of the json.load below

See also how things can be regenerated easily with a function used as a test help such as there https://github.com/gemfileparser/gemfileparser/pull/14/files#diff-245d8f4be70a2ec133e0c173d78ea046R49

Reply:
Okay just need some help here :sweat_smile: what should be done her to get rid of None values ? After this I can proceed with all the tests :)

Reply:
`data = {
        idx: packages[idx].to_dict()
            for idx in range(0,len(packages))
    }
`

Reply:
re: https://github.com/nexB/fetchcode/pull/31#discussion_r474192480

> Okay just need some help here sweat_smile what should be done here to get rid of None values ? After this I can proceed with all the tests :)

You keep the None alright. That's not a problem

Reply:
> `data = { str(idx): packages[idx].to_dict() for idx in range(0,len(packages)) } `

why do you need an index idx?  you do not need index in Python... ad in the rare you want one, use `enum
for idx in range(0,len(packages), here, I cannot fathom why you would need one. A simple list comprehension I suggested before is likely all you need.

Reply:
I thought to use list comprehension here, but I saw that we were generating lift of objects, but we needed dictionary of objects so I used indexes of list as key and package.to_dict() as value for that key.

Reply:
The list is perfectly fine and you are just doing an assert foo == bar so there is no need for more

Reply:
Thanks @pombredanne :smile: I have made the changes if they are good to go I can do same for others too

Reply:
Excellent: go for it. And create a helper function to keep things DRY https://en.wikipedia.org/wiki/Don't_repeat_yourself ;)